src/jit/codegenarm.cpp

```diff
@@ -1413,7 +1513,73 @@ void CodeGen::genFloatToFloatCast(GenTreePtr treeNode)
 //
 void CodeGen::genIntToFloatCast(GenTreePtr treeNode)
 {
-    NYI("Cast");
+    // int <--> float/double conversions are always non-overflow ones
+    assert(treeNode->OperGet() == GT_CAST);
+    assert(!treeNode->gtOverflow());
+
+    regNumber targetReg = treeNode->gtRegNum;
+    assert(genIsValidFloatReg(targetReg));
+
+    GenTreePtr op1 = treeNode->gtOp.gtOp1;
+    assert(!op1->isContained());             // Cannot be contained
+    assert(genIsValidIntReg(op1->gtRegNum)); // Must be a valid float reg.
+
+    var_types dstType = treeNode->CastToType();
+    var_types srcType = op1->TypeGet();
+    assert(!varTypeIsFloating(srcType) && varTypeIsFloating(dstType));
+
+    // force the srcType to unsigned if GT_UNSIGNED flag is set
+    bool isUnsignedSrc = ((treeNode->gtFlags & GTF_UNSIGNED) != 0);
+    if (isUnsignedSrc)
+    {
+        srcType = genUnsignedType(srcType);
+    }
+
+    // We should never see a srcType whose size is neither EA_4BYTE or EA_8BYTE
+    // For conversions from small types (byte/sbyte/int16/uint16) to float/double,
+    // we expect the front-end or lowering phase to have generated two levels of cast.
+    //
+    emitAttr srcSize = EA_ATTR(genTypeSize(srcType));
+    noway_assert((srcSize == EA_4BYTE) || (srcSize == EA_8BYTE));
+
+    genConsumeOperands(treeNode->AsOp());
+
+    instruction ins        = INS_invalid;
+    regMaskTP   tmpRegMask = treeNode->gtRsvdRegs;
+    regNumber   tmpReg     = genRegNumFromMask(tmpRegMask);
+    noway_assert(tmpReg != REG_NA);
```

Comment:
It doesn't look like you've added any code to tell LSRA to allocate a temporary register in TreeNodeInfoInit()